relayer/src/services/polygon-zkevm.ts

```diff
@@ -0,0 +1,322 @@
+      return message.claim_tx_hash;
+    }
+
+    // 4. Wait until message is ready to be claimed
+    console.log("Waiting for message to be ready for claiming.");
+    let readyForClaim = message.ready_for_claim;
+    while (!readyForClaim) {
+      const { ready_for_claim } = await this.getMessageInfoSafe(
+        bridgeEventParameters.depositCount,
+        bridgeEventParameters.originNetwork,
+      );
+
+      readyForClaim = ready_for_claim;
+
+      if (readyForClaim) {
+        break;
+      } else {
+        await sleep(5000);
+      }
+    }
+    console.log("Message ready to be claimed.");
+
+    // 5. Fetch merkle proof
+    const { proof } = await this.getMessageMerkleProof(
+      bridgeEventParameters.depositCount,
+      bridgeEventParameters.originNetwork,
+    );
+    console.log("Found MerkleProof, proceeding with the next steps...");
+
+    // 6. Claim message
+    const networkConfig = await this.getNetworkConfig();
+    const polygonBridgeContract = PolygonZKEvmBridge__factory.connect(
+      networkConfig.l1.PolygonZKEvmBridge,
+      this.l1Wallet,
+    );
+
+    const L1ClaimTxHash = await polygonBridgeContract.claimMessage(
+      proof.merkle_proof,
+      bridgeEventParameters.depositCount,
+      proof.main_exit_root,
+      proof.rollup_exit_root,
+      bridgeEventParameters.originNetwork,
+      bridgeEventParameters.originAddress,
+      bridgeEventParameters.destinationNetwork,
+      bridgeEventParameters.destinationAddress,
+      bridgeEventParameters.amount,
+      bridgeEventParameters.metadata,
+    );
+
+    console.log("Claim message transaction sent: ", L1ClaimTxHash.hash);
+
+    await L1ClaimTxHash.wait();
+    console.log("Successfully relayed message to L1.");
+
+    return L1ClaimTxHash.hash;
+  }
+
+  async finalize(l1TransactionHash: string): Promise<void> {
+    console.log("\n");
+    console.log("Claiming message travelling to PolygonZKEVM");
+
+    const receipt = await this.l1RpcProvider.waitForTransaction(l1TransactionHash, 1, 300000);
+
+    if (!receipt) {
+      throw new Error(`Transaction "${l1TransactionHash}" cannot be found on L1...`);
+    }
+    if (!receipt.status) {
+      throw new Error(`Transaction "${l1TransactionHash}" reverted on L1...`);
+    }
+
+    // 1. Extract bridge event parameters required for claiming a message
+    const bridgeEventParameters = await parseBridgeEvent(receipt.logs);
+    if (!bridgeEventParameters) {
+      throw new Error(
+        `Cannot find an event ("BridgeEvent") in the L1 transaction ${l1TransactionHash}`,
+      );
+    }
+    console.log("Found BridgeEvent. Proceeding with the next steps...");
+
+    // 2. Fetch message data
+    const message = await this.getMessageInfo(
+      bridgeEventParameters.depositCount,
+      bridgeEventParameters.originNetwork,
+    );
+
+    // 3. Check if message was already claimed on destination network
```

Comment:
my grammar check says it is more accurate to say: `check if the message was already claimed on the destination network`